src/DoctrineMessageRepository/DoctrineUuidV4MessageRepository.php

```diff
@@ -19,99 +19,35 @@
 
 class DoctrineUuidV4MessageRepository implements MessageRepository
 {
-    private int $jsonEncodeOptions = 0;
+    private DoctrineMessageRepository $messageRepository;
 
     public function __construct(
-        private Connection $connection,
-        private string $tableName,
-        private MessageSerializer $serializer,
-    ) {}
+        Connection $connection,
+        string $tableName,
+        MessageSerializer $serializer,
+    ) {
+        $this->messageRepository = new DoctrineMessageRepository(
+            connection: $connection,
+            serializer: $serializer,
+            tableSchema: new DefaultTableSchema(
+                tableName: $tableName
+            ),
+            uuidSerializer: new BinaryUuidSerializer(),
+        );
```

Comment:
Obviously this is a significant departure from what was previously happening here, but given that the previous constructor defined the properties as `private` it seems unlikely that anyone could extend this class in a useful way, so I doubt there is a compatibility issue.